### DISCUSSION TITLE
Unable to receive requests using HTTP DELETE/PUT/PATCH handlers

### DISCUSSION THREAD

**Discussion by Aynn** [uNetworking/uWebSockets.js#351] [Q&A] [answered]
Seems to work fine with the `any` method and I can see the appropriate method being listed under the `req.getHeader('access-control-request-method')` header but for some reason I can get these handlers work using their respective methods.

**Comment by hst-m** [comment]
Can you post the full code you are using? probably doing something wrong?

**Comment by Aynn** [comment]
Server (not working): 
```
  .del('/', (res, req) => {
    let origin =  req.getHeader('origin')
    res.writeHeader('Access-Control-Allow-Origin', origin);
    res.writeHeader('Access-Control-Allow-Credentials', 'true');    
    res.end();
  })
```
Server (working): 
```
  .any('/', (res, req) => {
    let origin =  req.getHeader('origin')
    res.writeHeader('Access-Control-Allow-Origin', origin);
    res.writeHeader('Access-Control-Allow-Credentials', 'true');    
    res.end();
  })
```
Client: 
```
    fetch('http://localhost:9001/', {
        method: 'DELETE',
    })
    .then(result=>{
        result.text().then(data=>console.log(data))
    })
    .catch(err=>console.log(err))
```

**Comment by Aynn** [comment]
Okay - Not sure what I was doing wrong but seems to be working fine now :| Thanks for the help! I will close this now.

**Comment by Aynn** [comment, marked as answer]
I will leave this comment here in case someone runs into a similar issue. 

I was testing my API through an application served on a different port (both localhost), this made it a cross site request.

Cross site requests get processed into two calls an options call to ensure the server accepts the type of request you're trying to make and then the actual call you intended to make. 

In order to to use non get/post routes I had to setup an `.options` route to with the appropriate `Access-Control-Allow-Origin` and `Access-Control-Allow-Methods`. 

Hope that helps!
